test/connectionUI.test.ts

```diff
@@ -83,21 +89,42 @@ suite('Connection UI tests', () => {
 			quickPickItemType: CredentialsQuickPickItemType.Mru,
 			label: undefined
 		};
-		let mockConnection = { connectionString: 'test' };
-		prompter.setup(p => p.promptSingle(TypeMoq.It.isAny())).returns(() => Promise.resolve(item));
-		prompter.setup(p => p.prompt(TypeMoq.It.isAny(), true)).returns(() => Promise.resolve(mockConnection));
-		return connectionUI.promptForConnection().then(() => {
-			connectionStore.verify(c => c.getPickListItems(), TypeMoq.Times.once());
-			prompter.verify(p => p.promptSingle(TypeMoq.It.isAny()), TypeMoq.Times.once());
-		});
+
+		let quickPickMock = TypeMoq.Mock.ofType<vscode.QuickPick<IConnectionCredentialsQuickPickItem>>();
+		quickPickMock.setup(q => q.items);
+		quickPickMock.object.items = [item];
+		quickPickMock.setup(q => q.show());
+		quickPickMock.setup(q => q.onDidChangeSelection((e) => [item] = e));
+		quickPickMock.object.onDidChangeSelection((e) => [item] = e);
+		quickPickMock.setup(q => q.onDidHide(() => false));
+		vscodeWrapper.setup(v => v.createQuickPick()).returns(() => quickPickMock.object);
+
+		connectionUI.promptForConnection();
```

Comment:
This returns a promise, you should be awaiting this call

Reply:
Do you have any ideas for how `promptForConnection` can be easier to test than it is at the moment? It this is awaited, then the function doesn't return. How can this pattern be made better?

https://github.com/microsoft/vscode-mssql/blob/228564b2d9c370539f1364a79cde661460e7a1d1/src/views/connectionUI.ts#L87